packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -179,13 +195,26 @@ export abstract class AbstractMessageManager<
 
   /**
    * Adds a passed Message to this.messages, and calls this.saveMessageList() to save
-   * the unapproved Messages from that list to this.messages.
+   * the unapproved Messages from that list to this.messages. If a security provider
+   * is provided, the method will check if the Message is malicious or not.
    *
    * @param message - The Message to add to this.messages.
    */
   addMessage(message: M) {
-    this.messages.push(message);
-    this.saveMessageList();
+    if (this.securityProviderRequest) {
```

Comment:
For the sake of clarity, could this go in a separate private method like `checkSecurity` which checks if a request was provided and simply returns the updated message, or the original one.

Reply:
Indeed make sense. I've added the `checkSecurity` and also the necessary tests.